in-progress/8754-slashing-and staking.md

```diff
@@ -0,0 +1,357 @@
+
+    alt If they have MIN_ACTIVATION_BALANCE
+        B ->> C: Make deposit at stake i.e. call activateValidator()
+        B ->> B: Add to validator set
+    else If they don't
+        B ->> B: Do nothing
+    end
+```
+
+### Migrating Stake
+
+The `Deposit` contract is owned by the Governance Contract. As such in the event of an upgrade, the Apella can change the `StakedDeposits` mapping so that any `StakeObject` with `followingRegistryBool=true` can be moved to the new Rollup as pointed to by the Registry contract. 
+
+Validators that choose `followingRegistryBool=false`, will have to exit funds and move manually. Note that this does not require block building on the old Rollup instance to be functional. 
+
+In the event of a Migration (i.e. state wipe), a new Deposit contract must be deployed. 
+
+### Withdraw Function
+
+* Validators can call the withdraw function at any time to remove any balance they have in `Deposits` using `withdraw()`.
+* Only the Rollup contract may withdraw balances from `stakedDeposits` using the function `unstakeValidator()`.
+
+### Rewards
+
+Any block rewards accruing to validators should be sent to the `withdrawalAddress` specified in `stakedDeposits`.
+
+# Proof Of Governance Slashing
+
+Why do we slash?
+
+We need to make sure the chain is always (eventually) finalizing new blocks. The conditions required for the chain to finalize new blocks are:
+1) Sequencers are making valid block proposals
+2) More than 2/3 of the committee is attesting on valid block proposals.
+3) Provers can obtain timely proof data for txs included in the epoch.
+4) Provers post proofs to the L1 on time.
+
+While Based Fallback acts as a backstop that ensures eventual liveness, malicious or permanently offline validators should eventually be kicked out of the validator set otherwise we end up in Based Fallback mode all too often. 
+
+In PoG, validators vote to slash any dishonest validators. To reduce coordination costs, we equip the validator client software with the ability to automatically initiate and respond to votes to slash dishonest validators based on onchain or offchain (i.e. p2p) evidence.
+
+Any validator can initiate a proposal to slash any other validator(s). A proposal to slash must include the offense, epoch/slot information and the validators to be slashed.  
+
+Validators vote by signing the slashing proposal and gossiping back the signed message over the p2p. Anyone can submit a slashing proposal that has gathered enough signatures to the L1 for verification. 
+
+Based on the offense in question, the rollup contract establishes max slashable stakes. The rollup contract also verifies certain conditions have occured before executing the slash. 
+
+This is more subjective than purely onchain slashing but it enables precision that is not possible with purely onchain slashing. We can adopt Ethereum's principle of "honest validators should not be slashed" at the expense of increased coordination cost.
+
```

Comment:
We will need to describe the mechanism / rollup contract function that allows slashing conditions to be added and their max slash-able stakes to be updated